src/algorithms/goto_algorithms.cpp

```diff
@@ -0,0 +1,197 @@
+#include <algorithms/goto_algorithms.h>
+#include <goto-programs/goto_loops.h>
+#include <goto-programs/remove_skip.h>
+
+void unwind_goto_functions::unroll_loop(
+  goto_programt &goto_program,
+  loopst &loop)
+{
+  get_loop_bounds bounds(goto_program, loop);
+  if(!bounds.run())
+    return;
+
+  number_of_bounded_loops++;
+  unsigned unwind = bounds.get_bound();
+  // TODO: What happens when K == K0? Should we optimize it out?
+  if(unwind == 0)
+    return;
```

Comment:
@rafaelsamenezes What happens if `bound` is non-deterministic? For example,
```c
size_t foo = nondet_size_t();
for(int i = 0; i < foo; ++i)
{
    /* ... */
}
```

Reply:
This case will be ignored, the bound extracting algorithm will return **false** which means that it isn't able to detect how many iteractions are needed.

In the future we could try to do some reasoning over the bounds and use some loop unrolling/transformation techniques.